### PR TITLE
PIN-4421: DtD reports job porting from Scala

### PR DESCRIPTION
**Pull request by Carminepo2**
This PR added a new job `metrics-report-generator` ported from Scala.

### CODE REVIEW

File: jobs/metrics-report-generator/src/utils/helpers.util.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+}
+
+export function generateAgreementsWorksheetTableData(
+  agreements: AgreementQueryData[],
+  purposes: PurposeQueryData[],
+  eservicesMap: Map<string, EServiceQueryData>,
+  tenantsMap: Map<string, TenantQueryData>
+): AgreementsWorksheetTableData[] {
+  return agreements.map<AgreementsWorksheetTableData>((agreement) => {
+    const agreementPurposes = purposes.filter((purpose) => purpose.eserviceId === agreement.eserviceId)
```
Comment (galales) [inline jobs/metrics-report-generator/src/utils/helpers.util.ts]: if I'm not missing something
```suggestion
    const agreementPurposes = purposes.filter((purpose) => purpose.eserviceId === agreement.eserviceId && purpose.consumerId === agreement.consumerId)
```

File: jobs/metrics-report-generator/src/utils/helpers.util.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    const consumer = tenantsMap.get(agreement.consumerId)
+    const producer = tenantsMap.get(agreement.producerId)
+    const eservice = eservicesMap.get(agreement.eserviceId)
+
+    if (!consumer) log.warn(`[Agreements Worksheet] Tenant consumer ${agreement.consumerId} not found in readmodel`)
+    if (!producer) log.warn(`[Agreements Worksheet] Tenant producer ${agreement.producerId} not found in readmodel`)
+    if (!eservice) log.warn(`[Agreements Worksheet] Eservice ${agreement.eserviceId} not found in readmodel`)
+
+    return AgreementsWorksheetTableData.parse({
```
Comment (galales) [inline jobs/metrics-report-generator/src/utils/helpers.util.ts]: is it required to parse data at runtime instead of using compile time check?
(same for other occurrences)

File: jobs/pn-consumers/src/index.ts [hunk 3]
```diff
@@ -54,7 +54,7 @@ async function main(): Promise<void> {
 
   await mailer.sendMail({
     from: env.SMTP_USER,
-    to: env.MAIL_RECIPIENTS,
+    to: env.MAIL_RECIPIENTS[0],
```
Comment (galales) [inline jobs/pn-consumers/src/index.ts]: why this change?

**Review by galales — CHANGES_REQUESTED** [review]

File: jobs/pn-consumers/src/index.ts (hunk 3 above)
Comment (Carminepo2) [reply jobs/pn-consumers/src/index.ts]: Oops, dev testing leftover

File: jobs/metrics-report-generator/src/utils/helpers.util.ts (hunk 2 above)
Comment (Carminepo2) [reply jobs/metrics-report-generator/src/utils/helpers.util.ts]: As I said in the other PR, you're right, this is not needed here (and probably not needed also in most of the other occurencies)

**Review by galales — APPROVED** [review]
❤️

**Comment by beetlecrunch** [comment]
This PR is no longer necessary.
Metrics will be generated by datalake.

**Comment by beetlecrunch** [comment]
Restored PR.
@Carminepo2
